docker/Dockerfile.quinoa-build-travis

```diff
@@ -31,23 +36,53 @@ WORKDIR build
 RUN ccache -z
 RUN cmake \
     ${COVERAGE:+-DCOVERAGE=ON} \
+    ${SONARQUBE:+-DTPL_DIR=/home/quinoa/tpl} \
     -DLCOV=/bin/true \
     -DGENHTML=/bin/true \
-    -DRUNNER_ARGS="" \
+    -DRUNNER_ARGS="${RUNNER_ARGS}" \
     -DCMAKE_BUILD_TYPE=Release \
     ../src
-RUN make VERBOSE=1 -j2
+RUN ${SONARQUBE:+build-wrapper-linux-x86-64 --out-dir bw-output} make VERBOSE=1 -j2
 RUN ccache -s
-RUN ./charmrun +p2 Main/unittest -v
-RUN ctest -j2 -E "(walker:SkewNormalPDF_pe8|walker:DiagOrnsteinUhlenbeckPDF_pe8)"
+RUN ./charmrun +p2 "${RUNNER_ARGS}" Main/unittest -v
+RUN ctest -j2 -LE stringent -E "(walker:SkewNormalPDF(.*)pe8|walker:DiagOrnsteinUhlenbeckPDF(.*)pe8)"
 RUN make install DESTDIR=${PWD}
 RUN cd .. && if [ ${COVERAGE} ]; then \
   if [ ${CC} = clang ]; then \
     $HOME/.local/bin/codecov --gcov-exec "llvm-cov gcov"; \
   else \
     $HOME/.local/bin/codecov; \
+    cd build && find . -type f -name '*.gcno'  -exec gcov -pb  {} + && \
+    find . -type f -name '*.gcov' | grep -v '#quinoa#src' | tr "\n" "\0" | xargs -0 rm && \
+    find . -type f -name '*.gcov' | grep 'def.h\|UnitTest#tests' | tr "\n" "\0" | xargs -0 rm; \
   fi; \
-fi && cd -
+fi
+RUN cd .. && if [ ${SONARQUBE} ]; then \
```

Comment:
I was thinking the same, but since sonarqube supports [branches](https://docs.sonarqube.org/display/SONAR/Local+and+Branch+Analysis), though [not in a perfect way](https://docs.sonarqube.org/display/SONAR/Analysis+Parameters#AnalysisParameters-ProjectConfiguration.1), I figured it is still useful. The disadvantage is that we have to and clean up old branches eventually - if we care enough about it. I'll keep the branches for now.